dotCMS/src/main/java/com/dotcms/rest/api/v1/template/TemplateResource.java

```diff
@@ -0,0 +1,752 @@
+        final Template currentTemplate = this.templateAPI.find(templateForm.getInode(), user, pageMode.respectAnonPerms);
+
+        if (null == currentTemplate || UtilMethods.isNotSet(currentTemplate.getIdentifier())
+                || !InodeUtils.isSet(currentTemplate.getInode())) {
+
+            throw new DoesNotExistException("The working template inode: " + templateForm.getInode() + " does not exists");
+        }
+
+        this.templateHelper.checkPermission(user, currentTemplate, PERMISSION_WRITE);
+
+        return Response.ok(new ResponseEntityView(this.templateHelper.toTemplateView(
+                this.fillAndSaveTemplate(templateForm, user, host, pageMode, currentTemplate), user))).build();
+    }
+
+    /**
+     * Publish a list of template inodes
+     * Return the list of a success published and failed
+     * @param request            {@link HttpServletRequest}
+     * @param response           {@link HttpServletResponse}
+     * @param templatesToPublish {@link List} list of template inodes to publish
+     * @return Response
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @PUT
+    @Path("/publish")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response publish(@Context final HttpServletRequest  request,
+                               @Context final HttpServletResponse response,
+                               final List<String> templatesToPublish) throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requestAndResponse(request, response).rejectWhenNoUser(true).init();
+        final User user         = initData.getUser();
+        final Host host         = this.hostWebAPI.getCurrentHostNoThrow(request);
+        final PageMode pageMode = PageMode.get(request);
+        final List<String> publishedInodes = new ArrayList<>();
+        final List<String> failedInodes    = new ArrayList<>();
+
+        if (!UtilMethods.isSet(templatesToPublish)) {
+
+            throw new IllegalArgumentException("The body must send a collection of template inodes such as: " +
+                    "[\"dd60695c-9e0f-4a2e-9fd8-ce2a4ac5c27d\",\"cc59390c-9a0f-4e7a-9fd8-ca7e4ec0c77d\"]");
+        }
+
+        for (final String templateInode : templatesToPublish) {
+
+            final Template template = this.templateAPI.find(templateInode, user, pageMode.respectAnonPerms);
+
+            if (null != template && InodeUtils.isSet(template.getInode())) {
+
+                try {
+
+                    if (this.templateAPI.publishTemplate(template, user, pageMode.respectAnonPerms)) {
+
+                        ActivityLogger.logInfo(this.getClass(), "Publish Template action", "User " +
+                                user.getPrimaryKey() + " publishing template" + template.getTitle(), host.getTitle() != null ? host.getTitle() : "default");
```

Comment:
Codacy found an issue: [The String literal "default" appears 4 times in this file; the first occurrence is on line 395](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6042990)